### PR TITLE
prepare 1.4.2 release

### PR DESCRIPTION
**Pull request by wochinge**
**Proposed changes**:
- prepare 1.4.2 release
- close https://github.com/RasaHQ/rasa/issues/4674
- close https://github.com/RasaHQ/rasa/issues/4643

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa_nlu#code-style) for instructions)

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -7,8 +7,8 @@ Rasa Change Log
 All notable changes to this project will be documented in this file.
 This project adheres to `Semantic Versioning`_ starting with version 1.0.
 
-[Unreleased 1.4.2]
-~~~~~~~~~~~~~~~~~~
+[1.4.2] - 2019-10-28
```
Comment (akelad) [inline CHANGELOG.rst]: can we add an "unreleased section" back in?

File: CHANGELOG.rst [hunk 2]
```diff
@@ -7,8 +7,8 @@ Rasa Change Log
...
+^^^^^^^^^^^^^^^^^^^^
 
 Added
```
Comment (akelad) [inline CHANGELOG.rst]: empty sections should go away in a release change log

File: CHANGELOG.rst (hunk 1 above)
Comment (wochinge) [reply CHANGELOG.rst]: done

File: CHANGELOG.rst (hunk 2 above)
Comment (wochinge) [reply CHANGELOG.rst]: ok, wasn't sure

File: CHANGELOG.rst [hunk 3]
```diff
@@ -7,8 +7,8 @@ Rasa Change Log
...
 -----
```
Comment (erohmensing) [inline CHANGELOG.rst]: remove added and changed please :grumpycat:

**Pull request merged**